client_test.go

```diff
@@ -152,23 +169,56 @@ func TestClient_Set(t *testing.T) {
 }
 
 func TestClient_GetByID(t *testing.T) {
-	Convey("Given a session ID client.GetByID returns a session", t, func() {
-		mockRedisClient, client := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringResult(string(resp), nil))
+	Convey("Given a session ID client.GetByID returns a session and TTL is refreshed", t, func() {
+		mockRedisClient, client := setUpMocks(
+			*redis.NewStatusCmd(),
+			*redis.NewStringResult(string(resp), nil),
+			*redis.NewStatusCmd(),
+			*redis.NewBoolCmd(),
+		)
 
 		Convey("When client uses the ID to get the session", func() {
 			s, err := client.GetByID("1234")
 			So(err, ShouldBeNil)
 
 			Convey("Then redis client.Get is called and returns the session", func() {
 				So(mockRedisClient.GetCalls(), ShouldHaveLength, 1)
+				So(mockRedisClient.ExpireCalls(), ShouldHaveLength, 1)
```

Comment:
Maybe it would be worth validating that the TTL was extended by the intended duration?
You could add the following extra check:
```
So(mockRedisClient.ExpireCalls()[0].Expiration, ShouldEqual, testTTL)
```

Reply:
Thanks David, I have now added this check. I was trying to figure out how to do this, so thank you 👍